### PR TITLE
[MNG-6999] Chained (consumer) XMLFilters can result in "floating" comments

### PR DESCRIPTION
**Pull request by rfscholte**

### CODE REVIEW

File: maven-core/src/test/java/org/apache/maven/internal/aether/ConsumerModelSourceTransformerTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import org.apache.maven.model.building.TransformerContext;
+import org.junit.Test;
+import org.xmlunit.assertj.XmlAssert;
+
+public class ConsumerModelSourceTransformerTest
+{
+    private ConsumerModelSourceTransformer transformer = new ConsumerModelSourceTransformer();
+
+    @Test
+    public void test() throws Exception
```
Comment (mthmulders) [inline maven-core/src/test/java/org/apache/maven/internal/aether/ConsumerModelSourceTransformerTest.java:37]: Would it be possible to come up with a name for the test that better describes what it verifies?

File: maven-xml/src/test/java/org/apache/maven/xml/sax/filter/ConsumerPomXMLFilterTest.java [hunk 2]
```diff
@@ -231,5 +234,22 @@ public void licenseHeader() throws Exception {
         String actual = transform( input );
         assertThat( actual ).and( expected ).areIdentical();
     }
+    
+    @Test
+//    @Ignore
```
Comment (mthmulders) [inline maven-xml/src/test/java/org/apache/maven/xml/sax/filter/ConsumerPomXMLFilterTest.java:239]: I think this comment can be removed :-)

File: maven-xml/src/test/java/org/apache/maven/xml/sax/filter/ConsumerPomXMLFilterTest.java [hunk 3]
```diff
@@ -231,5 +234,22 @@ public void licenseHeader() throws Exception {
...
+    public void lexicalHandler() throws Exception
```
Comment (mthmulders) [inline maven-xml/src/test/java/org/apache/maven/xml/sax/filter/ConsumerPomXMLFilterTest.java:240]: Should this test method not describe what it verifies, i.e. "removes modules" or something like that?

**Review by mthmulders — COMMENTED** [review]
I left a few remarks but I don't oversee what this code does, nor what the effect of the proposed changes would be.

File: maven-xml/src/test/java/org/apache/maven/xml/sax/filter/ConsumerPomXMLFilterTest.java (hunk 3 above)
Comment (rfscholte) [reply maven-xml/src/test/java/org/apache/maven/xml/sax/filter/ConsumerPomXMLFilterTest.java:240]: It tests that the lexical handler is used in both SaxSource and StreamResult. If one of both doesn't provide the lexical handler, all comments will be ignored. By default the SaxSource has no LexicalHandler, so the XMLParser will simply get elements, attributes, etc.

But yes, this is now testing 2 different things. I will add a test with a minimal pom + license, that should be enough to verify if the lexicalHander is properly chained.

File: maven-core/src/test/java/org/apache/maven/internal/aether/ConsumerModelSourceTransformerTest.java (hunk 1 above)
Comment (rfscholte) [reply maven-core/src/test/java/org/apache/maven/internal/aether/ConsumerModelSourceTransformerTest.java]: of course, my bad practice when starting a new class with only one method, to simply start with `test`

File: maven-core/src/test/java/org/apache/maven/internal/aether/ConsumerModelSourceTransformerTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    {
+        Path beforePomFile = Paths.get( "src/test/resources/projects/transform/before.pom").toAbsolutePath();
+        Path afterPomFile = Paths.get( "src/test/resources/projects/transform/after.pom").toAbsolutePath();
+
+        try( InputStream in = transformer.transform( beforePomFile, new NoTransformerContext() ) )
+        {
+            XmlAssert.assertThat( in ).and( Files.newInputStream( afterPomFile ) ).areIdentical();
```
[the remark this answers is not in the document]
Comment (rfscholte) [reply maven-core/src/test/java/org/apache/maven/internal/aether/ConsumerModelSourceTransformerTest.java]: no it does, good find!

File: maven-xml/src/test/java/org/apache/maven/xml/sax/filter/ConsumerPomXMLFilterTest.java (hunk 2 above)
Comment (rfscholte) [reply maven-xml/src/test/java/org/apache/maven/xml/sax/filter/ConsumerPomXMLFilterTest.java]: true

**Comment by rfscholte** [comment]
if there are no objections, I'll merge this tomorrow

**Review by MartinKanters — COMMENTED** [review]
I think the code looks good in general, but I added some very small comments, so it's up to you whether you want to process them or not.

File: maven-core/src/test/java/org/apache/maven/project/ProjectBuilderTest.java [hunk 5]
```diff
@@ -249,9 +249,9 @@ public void testReadInvalidPom()
         catch ( ProjectBuildingException ex )
         {
             assertEquals( 1, ex.getResults().size() );
-            MavenProject project = ex.getResults().get( 0 ).getProject();
-            assertNotNull( project );
-            assertNotSame( 0, ex.getResults().get( 0 ).getProblems().size() );
+            assertNotNull( ex.getResults().get( 0 ).getPomFile() );
```
[the remark this answers is not in the document]
Comment (rfscholte) [reply maven-core/src/test/java/org/apache/maven/project/ProjectBuilderTest.java:252]: Noted. Now that we are using Java 8 it should be quite to catch these leaks by using assertThrows. If you hit such a test class, just fix it as part of continuous maintenance.

**Pull request closed without merging**